src/hooks/useSafeState.ts

```diff
@@ -0,0 +1,33 @@
+// source: https://medium.com/technology-hits/how-to-fix-memory-leak-issue-in-react-js-using-hook-a5ecbf9becf8
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+type useStateSafeParams<T> = T | (() => T);
+type dispatch<T> = React.Dispatch<React.SetStateAction<T>>;
+
+/**
+ * Wrapper around react's `useState` hook.
+ * Use this hook to prevent memory leak as this wont call set state on unmounted component.
+ *
+ * @param initialValue initial state value
+ */
+export const useStateSafe = <T>(initialValue: useStateSafeParams<T>): [T, dispatch<T>] => {
```

Comment:
Hook is called useStateSafe, file is called useSafeState. Thus imported as useStateSafe from filename useSafeState. Would be nice to be consistent here.

Reply:
Ah, thanks for pointing it out: an oversight 😅 Will update.